### PR TITLE
[Proto Annotations] Parse out all Resources and ResourceSets

### PR DESCRIPTION
**Pull request by andreamlin**

### CODE REVIEW

File: src/main/java/com/google/api/codegen/util/ProtoParser.java [hunk 1]
```diff
@@ -46,6 +61,24 @@ public String getResourcePath(Field element) {
     return null;
   }
 
+  @SuppressWarnings("unchecked")
+  @Nullable
+  private <T, O extends Message, E extends ProtoElement> T getProtoExtension(
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/util/ProtoParser.java]: that's a gnarly signature!

File: src/main/java/com/google/api/codegen/util/ProtoParser.java [hunk 2]
```diff
@@ -61,21 +94,102 @@ public String getPackageName(Model model) {
...
+        if (Strings.isNullOrEmpty(baseName)) {
+          diagCollector.addDiag(
+              Diag.error(
+                  SimpleLocation.TOPLEVEL,
+                  "There is a %s option with"
+                      + " no name defined in proto file %s. %s.name is required.",
+                  fileExtension.getDescriptor().getFullName(),
+                  protoFile.getFullName(),
+                  fileExtension.getDescriptor().getFullName()));
+        }
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/util/ProtoParser.java]: should we `continue` here?

File: src/main/java/com/google/api/codegen/util/ProtoParser.java (hunk 1 above)
Comment (andreamlin) [reply src/main/java/com/google/api/codegen/util/ProtoParser.java]: action required? i just wanted to cram the internal proto extension parsing into a tiny private method that the outside world won't have to look at

File: src/main/java/com/google/api/codegen/util/ProtoParser.java (hunk 2 above)
Comment (andreamlin) [reply src/main/java/com/google/api/codegen/util/ProtoParser.java]: ooh good call

**Comment by andreamlin** [comment]
wait don't review yet; gotta fix CI first

File: src/main/java/com/google/api/codegen/util/ProtoParser.java (hunk 1 above)
Comment (pongad) [reply src/main/java/com/google/api/codegen/util/ProtoParser.java]: no action required, just took me a while decipher :)

**Comment by andreamlin** [comment]
Fixed CI
PTAL

**Pull request merged**